### PR TITLE
quantum++ 4.1 (new formula)

### PR DESCRIPTION
**Pull request by vsoftco**
Modern C++ quantum computing library

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/quantum++.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  desc "Modern C++ quantum computing library"
+  homepage "https://github.com/softwareQinc/qpp"
+  url "https://github.com/softwareQinc/qpp/archive/v4.1.tar.gz"
+  sha256 "b2d0f84a4726be71e45dc03ad0556ff22f7e6b92c19676391899e1df9091c061"
+  license "MIT"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
```
Comment (carlocab) [inline Formula/quantum++.rb:11]: Do you need this? i.e. does `brew livecheck quantum++` return wrong results without this?

File: Formula/quantum++.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      #include <qpp/qpp.h>
+
+      int main() {
+          using namespace qpp;
+          std::cout << disp(transpose(0_ket)) << std::endl;
+      }
+    EOS
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    assert_equal "1 0", shell_output("./build/qpp_test").squish
```
Comment (carlocab) [inline Formula/quantum++.rb:42]: Can we rewrite this to avoid use of `#squish`?

File: Formula/quantum++.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      #include <iostream>
+      #include <qpp/qpp.h>
+
+      int main() {
+          using namespace qpp;
+          std::cout << disp(transpose(0_ket)) << std::endl;
+      }
+    EOS
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
```
Comment (carlocab) [inline Formula/quantum++.rb:41]: I think it's worth considering whether we should just compile `qpp_test.cpp` without relying on CMake (so we don't need is as a `:test` dependency).

**Review by carlocab — COMMENTED** [review]
Thanks; could you squash your commits together too please?

File: Formula/quantum++.rb (hunk 1 above)
Comment (vsoftco) [reply Formula/quantum++.rb:11]: No, will remove and then squash

**Comment by vsoftco** [comment]
@carlocab Updated and squashed, thanks for the suggestion

**Review by carlocab — DISMISSED** [review]
Thanks!

File: Formula/quantum++.rb (hunk 3 above)
Comment (carlocab) [reply Formula/quantum++.rb:36]: That's not really possible. `cmake` will just call the compiler to build things, and we can always run the exact same command that `cmake` will.

**Comment by vsoftco** [comment]
@carlocab Thank you very much for all your help. Looks like all passes, didn't know I should give access to homebrew on my local branch, so looks like you need to re-approve the replaced commit :)

**Pull request closed without merging**